Sources/SourceKitLSP/Documentation/DocumentationLanguageService.swift

```diff
@@ -0,0 +1,296 @@
+import SwiftExtensions
+import SwiftParser
+import SwiftParserDiagnostics
+
+#if compiler(>=6)
+package import LanguageServerProtocol
+package import SKOptions
+package import SwiftSyntax
+package import ToolchainRegistry
+#else
+import LanguageServerProtocol
+import SKOptions
+import SwiftSyntax
+import ToolchainRegistry
+#endif
+
+package actor DocumentationLanguageService: LanguageService, Sendable {
+  package init?(
+    sourceKitLSPServer: SourceKitLSPServer,
+    toolchain: Toolchain,
+    options: SourceKitLSPOptions,
+    testHooks: TestHooks,
+    workspace: Workspace
+  ) async throws {}
+
+  package nonisolated func canHandle(workspace: Workspace) -> Bool {
+    return true
+  }
+
+  package func initialize(
+    _ initialize: InitializeRequest
+  ) async throws -> InitializeResult {
+    return InitializeResult(
+      capabilities: ServerCapabilities()
+    )
+  }
+
+  package func clientInitialized(_ initialized: InitializedNotification) async {
+    // Nothing to set up
+  }
+
+  package func shutdown() async {
+    // Nothing to tear down
+  }
+
+  package func addStateChangeHandler(
+    handler: @escaping @Sendable (LanguageServerState, LanguageServerState) -> Void
+  ) async {
+    // There is no underlying language server with which to report state
+  }
+
+  package func openDocument(
+    _ notification: DidOpenTextDocumentNotification,
+    snapshot: DocumentSnapshot
+  ) async {
+    // The DocumentationLanguageService does not do anything with document events
+  }
+
+  package func closeDocument(_ notification: DidCloseTextDocumentNotification) async {
+    // The DocumentationLanguageService does not do anything with document events
+  }
+
+  package func reopenDocument(_ notification: ReopenTextDocumentNotification) async {
+    // The DocumentationLanguageService does not do anything with document events
+  }
+
+  package func changeDocument(
+    _ notification: DidChangeTextDocumentNotification,
+    preEditSnapshot: DocumentSnapshot,
+    postEditSnapshot: DocumentSnapshot,
+    edits: [SwiftSyntax.SourceEdit]
+  ) async {
+    // The DocumentationLanguageService does not do anything with document events
+  }
+
+  package func willSaveDocument(_ notification: WillSaveTextDocumentNotification) async {
+    // The DocumentationLanguageService does not do anything with document events
+  }
+
+  package func didSaveDocument(_ notification: DidSaveTextDocumentNotification) async {
+    // The DocumentationLanguageService does not do anything with document events
+  }
+
+  package func documentUpdatedBuildSettings(_ uri: DocumentURI) async {
+    // The DocumentationLanguageService does not do anything with document events
+  }
+
+  package func documentDependenciesUpdated(_ uris: Set<DocumentURI>) async {
+    // The DocumentationLanguageService does not do anything with document events
+  }
+
+  package func completion(_ req: CompletionRequest) async throws -> CompletionList {
+    .init(isIncomplete: false, items: [])
```

Comment:
I generally have a dislike for `.init` initialization because it makes code harder to read because you don’t know which type is being initialized. Same a couple more times below.

```suggestion
    CompletionList(isIncomplete: false, items: [])
```

Reply:
Done. I also changed some of the other instances of `.init()` to use the structure name.